internal/controller/kbsconfig_controller.go

```diff
@@ -635,26 +637,118 @@ func (r *KbsConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		r.namespace = KbsOperatorNamespace
 	}
 
+	configMapMapper, err := configMapToKbsConfigMapper(r.Client, r.log)
+	if err != nil {
+		return err
+	}
+
+	secretMapper, err := secretToKbsConfigMapper(r.Client, r.log)
+	if err != nil {
+		return err
+	}
+
 	// Create a new controller and add a watch for KbsConfig including the following secondary resources:
 	// KbsConfigMap, KbsSecret, KbsAsConfigMap, KbsRvpsConfigMap in the same namespace as the controller
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&confidentialcontainersorgv1alpha1.KbsConfig{}).
 		// Watch for changes to ConfigMap, Secret that are in the same namespace as the controller
 		// The ConfigMap and Secret are not owned by the KbsConfig
 		Watches(
-			&source.Kind{Type: &corev1.ConfigMap{}},
-			&handler.EnqueueRequestForObject{},
+			&corev1.ConfigMap{},
+			handler.EnqueueRequestsFromMapFunc(configMapMapper),
 			builder.WithPredicates(namespacePredicate(r.namespace)),
 		).
 		Watches(
-			&source.Kind{Type: &corev1.Secret{}},
-			&handler.EnqueueRequestForObject{},
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(secretMapper),
 			builder.WithPredicates(namespacePredicate(r.namespace)),
 		).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
 		Complete(r)
+}
+
+// create mapper to transform from ConfigMap to KbsConfig
+func configMapToKbsConfigMapper(c client.Client, log logr.Logger) (handler.MapFunc, error) {
+	mapperFunc := func(ctx context.Context, o client.Object) []reconcile.Request {
+		log.Info("Got a configMap object")
```

Comment:
We do know it's configMap since this method is called in response to changes to any configMap in specific namespace. However there could be some errors in parsing. I think I should change the error message.
See my comment below to your other question.